src/js/components/DateInput/stories/Form.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Box, Button, DateInput, Form, FormField, Grommet } from 'grommet';
+import { grommet } from 'grommet/themes';
+
+const Example = () => {
+  const [value, setValue] = React.useState({ value: '' });
+  const onChange = nextValue => {
+    console.log('onChange', nextValue);
+    setValue(nextValue);
+  };
+  return (
+    <Grommet theme={grommet}>
+      <Box align="center" pad="large">
+        <Form
+          value={value}
+          onChange={onChange}
+          onSubmit={() => console.log(value)}
+        >
+          <FormField name="value" label="value" required>
+            <DateInput name="value" format="mm/dd/yyyy" />
+          </FormField>
+          <Button type="submit" label="submit" />
```

Comment:
It's impossible to reach the 'Submit' Button via keyboard navigation, DateInput drop keeps stealing the keyboard Tab action.

Reply:
yes, changed to handle Esc when focus is on MaskedInput and drop is open.